core/server/master/src/main/java/alluxio/master/meta/checkconf/ServerConfigurationChecker.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.meta.checkconf;
+
+import alluxio.PropertyKey;
+import alluxio.PropertyKey.Scope;
+import alluxio.PropertyKey.ConsistencyCheckLevel;
+import alluxio.wire.Address;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class is responsible for checking server-side configuration.
+ */
+public class ServerConfigurationChecker {
+  /** Contain all the master configuration information. */
+  private final ServerConfigurationRecord mMasterRecord;
+  /** Contain all the worker configuration information. */
+  private final ServerConfigurationRecord mWorkerRecord;
+  /** Contain the checker results. */
+  private ConfigCheckReport mConfigCheckReport;
+
+  /**
+   * Status of the check.
+   */
+  public enum Status {
+    PASSED, // do not have configuration errors and warnings
+    WARN, // do not have configuration errors but have warnings
+    FAILED, // have configuration errors
+    NOT_STARTED,
+  }
+
+  /**
+   * Represents a configuration report which records the configuration checker results.
+   * Since we check server-side configuration, Scope here only includes
+   * SERVER, MASTER and WORKER. Scope.ALL will be considered as Scope.SERVER.
+   */
+  public static final class ConfigCheckReport {
+    /** Record the configuration errors. */
+    private final Map<Scope, List<WrongProperty>> mConfigErrors;
+    /** Record the configuration warnings. */
+    private final Map<Scope, List<WrongProperty>> mConfigWarns;
+    /**
+     * Record the status of masters and workers.
+     * Scope here includes MASTER and WORKER and their corresponding Statuses represent
+     * whether errors/warnings exist in masters/workers.
+     */
+    private final Map<Scope, Status> mConfigStatus;
```

Comment:
is it necessary to keep track of this?

Reply:
Deleted mConfigStatus. This Config Status makes the report too complicated but does not actually do something to the users.